database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -85,8 +85,32 @@ public String parseSnapshot(DataSnapshot snapshot) {
                 return snapshot.getKey();
             }
         });
+    }
+
+    @Override
+    public ChangeEventListener addChangeEventListener(@NonNull ChangeEventListener listener) {
+        boolean wasListening = isListening();
+        super.addChangeEventListener(listener);
+
+        // Only start listening when the first listener is added
+        if (!wasListening) {
+            mKeySnapshots.addChangeEventListener(this);
+        }
 
-        mKeySnapshots.addChangeEventListener(this);
```

Comment:
And on top of that, we would start listening immediately when the class was initialized instead of waiting for the first listener like we properly do in `FirebaseArray`. Double ouch! 😱